go/store/nbs/table_index_test.go

```diff
@@ -56,6 +56,62 @@ func TestParseTableIndex(t *testing.T) {
 	}
 }
 
+func TestParseLargeTableIndex(t *testing.T) {
+	// This is large enough for the NBS table index to overflow uint32s on certain index calculations.
+	numChunks := uint32(320331063)
+	idxSize := indexSize(numChunks)
+	sz := idxSize + footerSize
+	idxBuf := make([]byte, sz)
+	copy(idxBuf[idxSize+12:], magicNumber)
+	binary.BigEndian.PutUint32(idxBuf[idxSize:], numChunks)
+	binary.BigEndian.PutUint64(idxBuf[idxSize+4:], uint64(numChunks)*4*1024)
+
+	var prefix uint64
+
+	off := 0
+	// Write Tuples
+	for i := uint32(0); i < numChunks; i++ {
+		binary.BigEndian.PutUint64(idxBuf[off:], prefix)
+		binary.BigEndian.PutUint32(idxBuf[off+hash.PrefixLen:], i)
+		prefix += 2
+		off += prefixTupleSize
+	}
+
+	// Write Lengths
+	for i := uint32(0); i < numChunks; i++ {
+		binary.BigEndian.PutUint32(idxBuf[off:], 4*1024)
+		off += lengthSize
+	}
+
+	// Write Suffixes
+	for i := uint32(0); i < numChunks; i++ {
```

Comment:
The three 320M iteration loops raises my eyebrows. How long does this take?

Reply:
And how much memory does this use? Can this be precomputed and checked in?

Reply:
It takes a few seconds to run. I'm not sure how much memory it uses. Somewhere between 4GB and 8GB.